tests/ad-tests.dx

```diff
@@ -439,3 +439,18 @@ custom-linearization h hLin
 
 deriv (\x. h x x) 1.0 == deriv (\x. x + x) 1.0
 > True
+
+-- Index-polymorphic custom linearization
+def w {n} [Ix n] (x:n=>Float) : n=>Float = 2 .* x
+def wLin {n} [Ix n] (x:n=>Float) : (n=>Float & n=>Float -> n=>Float) = (w x, \xt:(n=>Float). 4 .* xt)
+
+custom-linearization w wLin
+
+jvp (\x. w x) [1.0, 1.0] [1.0, 0.0]
+> [4., 0.]
+
+-- This is a little unexpected at first, but the result makes sense when you think about it...
+-- w has implicit args, so we're really taking a jvp of (w _ _), which is a function returned by
+-- a partial application of w to some inference holes.
+jvp w [1.0, 1.0] [1.0, 0.0]
+> [2., 0.]
```

Comment:
I don't understand this comment.  Does this mean that the implicit partial application is defeating the custom linearization?  If so, that seems like a serious bug.  Otherwise, why is the answer not 4?

Reply:
Resolved in offline conversation by filing #990.